transport/header.go

```diff
@@ -20,41 +20,76 @@
 
 package transport
 
-import "strings"
+import "github.com/yarpc/yarpc-go/internal"
 
-// Headers represents YARPC headers.
+// CanonicalizeHeaderKey canonicalizes the given header key for storage into
+// the Headers map.
+func CanonicalizeHeaderKey(k string) string {
+	return internal.CanonicalizeHeaderKey(k)
+}
+
+// Headers is the transport-level representation of application headers.
 //
-// Header keys are case insensitive and duplicates are disallowed.
-type Headers map[string]string
+// Keys in the map MUST be canonicalized with CanonicalizeHeaderKey.
+//
+// You probably want to look at yarpc.Headers instead.
+type Headers internal.Headers
```

Comment:
I'm not clear why we need to re-expose the header func down in the transport package. Is the `yarpc` and `internal` package not enough?